src/Mod/TechDraw/TechDrawTools/TaskFillTemplateFields.py

```diff
@@ -0,0 +1,462 @@
+"""Provides the TechDraw FillTemplateFields Task Dialog."""
+
+__title__ = "TechDrawTools.CommandFillTemplateFields"
+__author__ = "Syres"
+__url__ = "https://www.freecad.org"
+__version__ = "00.02"
+__date__ = "2023/12/07"
+
+from PySide.QtCore import QT_TRANSLATE_NOOP
+from PySide import QtCore, QtGui
+from PySide.QtGui import QLineEdit, QCheckBox
+import FreeCAD as App
+import FreeCADGui as Gui
+import datetime
+from datetime import date
+import csv
+import codecs
+import os.path
+
+CreatedByChkLst = []
+ScaleChkLst = []
+LabelChkLst = []
+CommentChkLst = []
+CompanyChkLst = []
+LicenseChkLst = []
+CreatedDateChkLst = []
+LastModifiedDateChkLst = []
+listofkeys = [
+    "CreatedByChkLst",
+    "ScaleChkLst",
+    "LabelChkLst",
+    "CommentChkLst",
+    "CompanyChkLst",
+    "LicenseChkLst",
+    "CreatedDateChkLst",
+    "LastModifiedDateChkLst",
+]
+
+
+"""Run the following code when the command is activated (button press)."""
+file_path = App.getResourceDir() + "Mod/TechDraw/CSVdata/FillTemplateFields.csv"
+
+if os.path.exists(file_path):
+    with codecs.open(file_path, encoding="utf-8") as fp:
+        reader = csv.DictReader(fp)
+        if listofkeys == reader.fieldnames:
+            for row in reader:
+                CreatedByChkLst.append(row["CreatedByChkLst"])
+                ScaleChkLst.append(row["ScaleChkLst"])
+                LabelChkLst.append(row["LabelChkLst"])
+                CommentChkLst.append(row["CommentChkLst"])
+                CompanyChkLst.append(row["CompanyChkLst"])
+                LicenseChkLst.append(row["LicenseChkLst"])
+                CreatedDateChkLst.append(row["CreatedDateChkLst"])
+                LastModifiedDateChkLst.append(row["LastModifiedDateChkLst"])
+        else:
+            errorMsg = QtCore.QT_TRANSLATE_NOOP(
+                "Techdraw_FillTemplateFields",
+                " file does not contain the correct field names therefore exiting",
+            )
+            App.Console.PrintError(file_path + errorMsg + "\n")
+else:
+    errorMsg = QtCore.QT_TRANSLATE_NOOP(
+        "Techdraw_FillTemplateFields",
+        " file has not been found therefore exiting",
+    )
+    App.Console.PrintError(file_path + errorMsg + "\n")
+
+keyLst = []
+
+
+class TaskFillTemplateFields:
+    def __init__(self):
+        objs = App.ActiveDocument.Objects
+        for obj in objs:
+            if (
+                obj.TypeId == "TechDraw::DrawPage"
+                and os.path.exists(file_path)
+                and listofkeys == reader.fieldnames
+            ):
+                self.page = obj
+                if obj.Views == []:
+                    msgBox = QtGui.QMessageBox()
+                    msgTitle = QtCore.QT_TRANSLATE_NOOP(
+                        "Techdraw_FillTemplateFields",
+                        "View or Projection Group missing",
+                    )
+                    msg = QtCore.QT_TRANSLATE_NOOP(
```

Comment:
Might be different in Python, but AFAIK:

QT_TRANSLATE_NOOP(context, stringToTranslate) is for marking strings for translation so they
get picked up and sent to CrowdIn for the translators

QCoreApplication.translate(context, stringToTranslate) does the actual translating by searching the 
translation files and returning the translated version of stringToTranslate